types/result.go

```diff
@@ -119,7 +119,7 @@ func (r TxResponse) String() string {
 	}
 
 	if r.Log != "" {
-		sb.WriteString(fmt.Sprintf("  Log: %s\n", r.Log))
+		sb.WriteString(fmt.Sprintf("  Log: \n    %s\n", r.Log))
```

Comment:
++

Reply:
ohh hang on - am I right in thinking that this would unconditionally lead to multiline log messages? If so, please don't - it's notoriously hard to parse log multi-line entries...

Reply:
It already is multi-line, see: `BaseApp#runMsgs`

```go
	result = sdk.Result{
              // ...
		Log:       strings.Join(logs, "\n"),
              // ...
	}
```

Reply:
jeez...

Reply:
looks like you have a similar desire to @faboweb in wanting abci.Log to be human-readable structured data opposed to just a string. Unfortunately, this is ABCI breaking and requires changes in TM.

Reply:
Actually thinking about it some more, I think this can just be a JSON array!